### ISSUE TITLE
--extract shows errors past end of file

### ISSUE THREAD

**Issue by Arcanemagus** [jshint/jshint#2894]
While testing https://github.com/AtomLinter/linter-jshint/issues/232#issuecomment-194206998 on `jshint@2.9.1` I found that this was another case of `--extract` messing something up.

For some reason with the file provided in the bug report it is showing a line of 100, when the file only has 85 lines.

I've managed to cut the example down to the following code to trigger this:

```html
<script>
  a({
    b: { c: {}
    },
    function d(){};
  });
</script>

```
![image](https://cloud.githubusercontent.com/assets/427137/13712619/868822fe-e779-11e5-83f5-d4a945147ea0.png)
[task-list-template.zip](https://github.com/jshint/jshint/files/169782/task-list-template.zip)

As you can see it is reporting errors on line 9, when the file only has 8 lines. This only triggers if the file is `CRLF`, switching it to `LF` works around whatever bug this is.

**Comment by nicolo-ribaudo** [comment]
I think this is fixed by #2828

**Comment by Arcanemagus** [comment]
That looks likely, especially since switching the file to `LF` makes it report a sane line number. For now I've pushed out a version of `linter-jshint` that caps the line when `jshint` gives it invalid data.

**Comment by jugglinmike** [comment]
Thanks for the report! This has been resolved via gh-2828. You can expect the fix in the next (post-2.9.1) release of JSHint

**Comment by Arcanemagus** [comment]
Excellent! I think there are a few workarounds in there that this next release will mean we can remove :wink:.
